src/mediasession-helper.js

```diff
@@ -56,7 +59,19 @@ export function setMediaSession(win, metadata, playHandler, pauseHandler) {
     navigator.mediaSession.setActionHandler('pause', pauseHandler);
 
     // TODO(@wassgha) Implement seek & next/previous
+
+    const currentMediaSession = ++lastMediaSession;
+    return () => {
+      if (currentMediaSession !== lastMediaSession) {
+        return;
+      }
+
+      navigator.mediaSession.metadata = null;
+      navigator.mediaSession.setActionHandler('play', null);
+      navigator.mediaSession.setActionHandler('pause', null);
+    };
   }
+  return () => {};
```

Comment:
Totally fine to do a contract of `!Unlisten|undefined`. I think semantically it's a lot better. Literally means "this effect invocation has no cleanup". Also, in Preact, an empty function might schedule an extra cleanup pass, while `undefined` wouldn't.

Reply:
Good to know, changed.